### PR TITLE
Add support for displaying variation options in SKU Selector as a slider

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

Some stores would have products with too many SKUs, and that would make them regret opting to use SKU inside a `product-summary` block.

This PR aims to address that by enabling users to set the `displayMode` prop from SKU Selector to `"slider"` so that after a certain number of options each SKU variation is displayed as a slider inside the `product-summary`.

#### How should this be manually tested?

[Workspace](https://skuselectorslide--storecomponents.myvtex.com/top?_q=top&map=ft)

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

Depends on: https://github.com/vtex-apps/slider-layout/pull/13

### CODE REVIEW

File: react/__mocks__/vtex.responsive-values/index.js [hunk 1]
```diff
@@ -1 +1,2 @@
 export const useResponsiveValues = values => values
+export const useResponsiveValue = values => values
```
Comment (iaronaraujo) [inline react/__mocks__/vtex.responsive-values/index.js]: Are these both necessary?

File: react/components/SKUSelector/index.tsx [hunk 2]
```diff
@@ -199,20 +223,24 @@ const SKUSelectorContainer: FC<Props> = ({
...
+    setSelectedVariations(
+      getNewSelectedVariations(query, skuSelected, variations, initialSelection)
+    )
   }, [variations])
 
   useEffect(() => {
     if (skuSelected && onSKUSelected) {
       onSKUSelected(skuSelected.itemId)
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (iaronaraujo) [inline react/components/SKUSelector/index.tsx]: I think it is better to remove this comment even if it will result in a warning

Comment (iaronaraujo) [reply react/components/SKUSelector/index.tsx]: And maybe add a comment explaining why it is ok to do what you are doing even though it results in a warning

File: react/components/SKUSelector/index.tsx [hunk 3]
```diff
@@ -277,6 +305,7 @@ const SKUSelectorContainer: FC<Props> = ({
         }
       }
     },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (iaronaraujo) [inline react/components/SKUSelector/index.tsx]: same

File: react/__mocks__/vtex.responsive-values/index.js (hunk 1 above)
Comment (victorhmp) [reply react/__mocks__/vtex.responsive-values/index.js]: Yep, they do slightly different things. 
https://github.com/vtex-apps/responsive-values
`useResponsiveValues` is used to process an array of values, while `useResponsiveValue` is for a single one

Comment (lbebber) [inline react/__mocks__/vtex.responsive-values/index.js]:
```suggestion
export const useResponsiveValue = value => value
```

:drake-yes:

File: react/components/SKUSelector/Wrapper.tsx [hunk 4]
```diff
@@ -51,21 +66,26 @@ interface Props {
...
   thumbnailImage?: string
   visibleVariations?: string[]
   showVariationsLabels?: boolean
   variationsSpacing?: number
   showVariationsErrorMessage?: boolean
   initialSelection?: InitialSelectionType
-  displayMode?: DisplayMode
+  displayMode?: MaybeResponsiveInput<DisplayMode>
+  sliderDisplayThreshold?: number
+  sliderArrowIconSize?: number
```
Comment (lbebber) [inline react/components/SKUSelector/Wrapper.tsx]: :pick: acho q `sliderArrowSize` já é o suficiente

File: react/components/SKUSelector/Wrapper.tsx [hunk 5]
```diff
@@ -51,21 +66,26 @@ interface Props {
   hideImpossibleCombinations?: boolean
   showValueNameForImageVariation?: boolean
   showValueForVariation?: ShowValueForVariation
-  imageHeight?: number | object
-  imageWidth?: number | object
+  imageHeight?: MaybeResponsiveInput<number>
+  imageWidth?: MaybeResponsiveInput<number>
```
Comment (lbebber) [inline react/components/SKUSelector/Wrapper.tsx]: 👌

File: react/components/SKUSelector/index.tsx (hunk 2 above)
Comment (lbebber) [reply react/components/SKUSelector/index.tsx]: I agree with explaining why you are doing this, but keep the eslint-disable, so auto-formatters won't change it back

File: react/components/SKUSelector/index.tsx (hunk 3 above)
Comment (lbebber) [reply react/components/SKUSelector/index.tsx]: same

File: react/__mocks__/vtex.responsive-values/index.js (hunk 1 above)
Comment (victorhmp) [reply react/__mocks__/vtex.responsive-values/index.js]: hehe

**Comment by victorhmp** [comment]
> Could you provide an example in the product page?

Sure! https://skuselectorslide--storecomponents.myvtex.com/star-color-top/p?skuId=2000564

**Pull request merged**